samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -170,7 +179,30 @@ public String getSideInputsProcessorFactory(String storeName) {
    * @param storeName name of the store
    * @return the serialized instance of SideInputsProcessor if present, null otherwise
```

Comment:
Would be better to update the java doc to indicate that this method returns Optional.empty if the side-input factory is not defined.